testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/example/cors/AbstractCorsExampleAdapterTest.java

```diff
@@ -117,12 +119,32 @@ public void angularCorsProductTest() {
         angularCorsProductPage.loadPublicRealmInfo();
         waitUntilElement(angularCorsProductPage.getOutput()).text().contains("Realm name: cors");
 
-        Client client = ClientBuilder.newClient();
-        WebTarget target = client.target(authServerPage.createUriBuilder()).path("version");
-        VersionRepresentation version = target.request().get(VersionRepresentation.class);
+        String serverVersion = getAuthServerVersion();
 
+        assertNotNull(serverVersion);
+
+        angularCorsProductPage.navigateTo();
+        WaitUtils.pause(2000);
         angularCorsProductPage.loadVersion();
-        waitUntilElement(angularCorsProductPage.getOutput()).text().contains("Keycloak version: " + version.getVersion());
+        waitUntilElement(angularCorsProductPage.getOutput()).text().contains("Keycloak version: " + serverVersion);
+    }
+
+    @Nullable
+    private String getAuthServerVersion() {
+        driver.navigate().to(suiteContext.getAuthServerInfo().getContextRoot().toString() +
+                "/auth/admin/master/console/#/server-info");
+        testRealmLoginPage.form().login("admin", "admin");
```

Comment:
I think there might be a problem with htmlunit on this test. I think we need to rewrite it so that those tests were using Javascript browser, because Javascript adapter doesn't work with htmlunit (admin console uses Javascript Adapter).

We need to do it similarly to this:
https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/javascript/JavascriptAdapterTest.java

which is using driver annotated with @JavascriptBrowser
https://github.com/keycloak/keycloak/blob/master/testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adapter/javascript/AbstractJavascriptTest.java#L54

I can do it for you if you want. I am working on making all tests which are using Javascript adapter to use driver annotated with @JavascriptAdapter and I have forgotten about CORS tests. So we can merge this like it is and I will update it.

Reply:
+1. I appreciate your help here.